docs/external-node.md

```diff
@@ -0,0 +1,569 @@
+example,
+
+```yaml
+policyBypassRules:
+  - direction: ingress
+    protocol: tcp
+    cidr: 1.1.1.1/32
+    port: 22
+```
+
+The `direction` in one policyBypassRule supports `ingress` or `egress`. And
+the supported protocols include, `tcp`,`udp`, `icmp` and `ip`. The `cidr`
+gives the peer address, it is the destination in an `egress` rule, and the
+source in an `ingress` rule. For `tcp` and `udp` protocols, the `port` is
+required to describe the destination port in the packet.
+
+The policyBypassRules are processed before Antrea NetworkPolicy rules when
+forwarding the packets in OVS datapath, and the disposition of the rules is to
+allow the packets.
+
+## OpenFlow pipeline
+
+A new OpenFlow pipeline is implemented by `antrea-agent` dedicated for
+`ExternalNode` feature.
+
+![OVS pipeline](assets/ovs-pipeline-external-node.svg)
+
+### Non-IP packet
+
+`NonIPTable` is a new OpenFlow table introduced only when on an external Node,
+which is dedicated for all non-IP packets. Non-IP packet is forwarded between
+the pair ports directly, e.g., a non-IP packet entering OVS from the uplink
+interface is output to the paired internal port, and a packet from the internal
+port is output to the uplink.
+
+### IP packet
+
+One OpenFlow pipeline is set on an external Node to process the IP packet.
+Antrea NetworkPolicy is the major function in this new pipeline, and the
+OpenFlow tables used are similar to the container case. **No L3 routing is
+provided on an external Node**, and a simple L2 forwarding policy is
+implemented. OVS connection track is used to assist the NetworkPolicy function,
+in a result only the first packet is validated by the OpenFlow entries, and the
+sequencing packets in an accepted connection are allowed directly.
+
+- Egress/Ingress Tables
+
+Table `XgressSecurityClassifierTable` is installed in both `stageEgressSecurity`
+and `stageIngressSecurity`, which is used to install the OpenFlow entries for
+the [`policyBypassRules`](#bypass-antrea-networkpolicy) in the agent configuration.
+This is an example of the OpenFlow entry for the above configuration,
+
+```yaml
+table=IngressSecurityClassifier, priority=200,ct_state=+new+trk,tcp,nw_src=1.1.1.1,tp_dst=22 actions=resubmit(,IngressMetric)
+```
+
+Other OpenFlow tables in `stageEgressSecurity` and `stageIngressSecurity` are
+the same as those installed on Kubernetes worker Node. For more details about
+these tables, please refer to the general [introduction](design/ovs-pipeline.md)
+of Antrea OVS pipeline.
+
+- L2 Forwarding Tables
+
+`L2ForwardingCalcTable` is used to calculate the expected output port of an IP
+packet. As the pair ports with the internal port and uplink always exist on the
+OVS bridge, and both interfaces are configured with the same MAC address, the
+match condition of the OpenFlow entry in `L2ForwardingCalcTable` is the input
+port number but not the MAC address of the packet. The actions include,
```

Comment:
The flow actions?

include -> are?

, -> :

Reply:
updated.